libevmasm/Assembly.cpp

```diff
@@ -98,14 +98,23 @@ void Assembly::addAssemblyItemsFromJSON(Json::Value const& _code)
 
 AssemblyItem Assembly::createAssemblyItemFromJSON(Json::Value const& _json)
 {
-	std::string name = _json["name"].isString() ? _json["name"].asString() : "";
-	int begin = _json["begin"].isInt() ? _json["begin"].asInt() : -1;
-	int end = _json["end"].isInt() ?  _json["end"].asInt() : -1;
-	int srcIndex = _json["source"].isInt() ? _json["source"].asInt() : -1;
-	size_t modifierDepth = _json["modifierDepth"].isInt() ? static_cast<size_t>(_json["modifierDepth"].asInt()) : 0;
-	std::string value = _json["value"].isString() ? _json["value"].asString() : "";
-	std::string jumpType = _json["jumpType"].isString() ? _json["jumpType"].asString() : "";
-	solAssert(!name.empty(), "");
+	solAssert(ofType<std::string>(_json, "name"));
+	solAssert(ofType<int>(_json, "begin"));
+	solAssert(ofType<int>(_json, "end"));
+	solAssert(ofType<int>(_json, "source"));
+	solAssert(ofTypeIfExists<std::string>(_json, "value"));
+	solAssert(ofTypeIfExists<int>(_json, "modifierDepth"));
+	solAssert(ofTypeIfExists<std::string>(_json, "jumpType"));
+
+	std::string name = getOrDefault<std::string>(_json, "name", "");
+	solAssert(!name.empty());
+
+	int begin = getOrDefault<int>(_json, "begin", -1);
```

Comment:
Theoretically we only need a `get` function for the mandatory elements here (`name`, `begin`, `end` and `source`).

Reply:
I added now a function `get`. I think its more consistent to have this function together with `getOrDefault`.